src/ServiceManagement/Profile/Commands.Profile/ServicePrincipal/RemoveAzureSubscriptionServicePrincipal.cs

```diff
@@ -0,0 +1,44 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using Microsoft.Azure;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using Microsoft.WindowsAzure.Management;
+
+
+namespace Microsoft.WindowsAzure.Commands.Profile
+{
+    [Cmdlet(VerbsCommon.Remove, AzureSubscriptionServicePrincipalNounName)]
+    [OutputType(typeof(AzureOperationResponse))]
+    public class RemoveAzureSubscriptionServicePrincipal : ServiceManagementBaseCmdlet
+    {
+        public RemoveAzureSubscriptionServicePrincipal()
+            : base()
+        {
+
+        }
+
+        [Parameter(Position = 0, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "ServicePrincipal Object Id")]
```

Comment:
No Value from pipeline by propertyname, you should take direct pipeline input to a parameter of type service principal, and have a separate parameter set for objectid.  See: https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/azure-powershell-design-guidelines.md#inputobject-parameter-set

Reply:
I have removed Piping, as there is no usecase for it